### PR TITLE
Update kubeflow components to v0.4.0

### PR DESCRIPTION
**Pull request by richardsliu**
Updating the following kubeflow components to v0.4.0:
- tf-operator
- pytorch-operator
- centraldashboard
- Jupyter notebook images

Katib was recently updated (see https://github.com/kubeflow/kubeflow/pull/2064) so it is not included here.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2112)

### CODE REVIEW

**Comment by johnugeorge** [comment]
@richardsliu  image name seems to be wrong 

pytorch_operator -> pytorch-operator

File: kubeflow/pytorch-job/prototypes/pytorch-operator.jsonnet [hunk 1]
```diff
@@ -5,7 +5,7 @@
 // @param name string Name to give to each of the components
 // @optionalParam disks string null Comma separated list of Google persistent disks to attach to jupyter environments.
 // @optionalParam cloud string null String identifying the cloud to customize the deployment for.
-// @optionalParam pytorchJobImage string gcr.io/kubeflow-images-public/pytorch-operator:v0.3.0-16-g8d71ff3 The image for the PyTorchJob controller
+// @optionalParam pytorchJobImage string gcr.io/kubeflow-images-public/pytorch_operator:v0.4.0 The image for the PyTorchJob controller
```
Comment (johnugeorge) [inline kubeflow/pytorch-job/prototypes/pytorch-operator.jsonnet]: pytorch_operator -> pytorch-operator

Comment (richardsliu) [reply kubeflow/pytorch-job/prototypes/pytorch-operator.jsonnet]: Thanks, fixed.

**Pull request merged**